doc/SONiC_VRRP_HLD.md

```diff
@@ -0,0 +1,582 @@
+
+# Revision
+| Rev  |    Date    |       Author        | Change Description                                           |
+|:--:|:--------:|:-----------------:|:------------------------------------------------------------:|
+| 0.1  | 09/28/2019 |   Dilip Kumar (Broadcom)   | Initial version                                              |
+|      |            |             |                    |
+
+# Definition/Abbreviation
+
+### Table 1: Abbreviations
+
+| **Term**      | **Meaning**                                                  |
+| ------------- | ------------------------------------------------------------ |
+| ARP           | Address Resolution Protocol                                  |
+| CLI           | Command Line Interface                                       |
+| IP            | Internet Protocol                                            |
+| LAG           | Link Aggregation Group                                       |
+| LAN           | Local Area Network                                           |
+| MAC           | Media Access Control addresses                               |
+| VIP           | Virtual IP address                                           |
+| VLAN          | Virtual Local Area Network                                   |
+| VMAC          | Virtual MAC address                                          |
+| VRF           | Virtual Routing and Forwarding                               |
+| VRID          | Virtual Router Identifier                                    |
+| VRRP          | Virtual Router Redundency Protocol                           |
+| VRRP Instance | An instance of VRRP state machine on an interface. Multiple VRRP state machines can be configured on an interface. |
+| VRRP Owner    | VRRP owner (of a VRRP instance) is the router whose virtual IP address is the same as the real interface IP address |
+
+# About this Manual
+
+This document provides general overview of VRRP feature implementation based on RFC 3768 in SONiC. 
+
+
+
+# 1 Introduction and Scope
+
+This document describes the Functionality and High level design of the VRRP feature.
+
+The Virtual Router Redundancy Protocol (VRRP) functionality is designed to eliminate the single point of
+failure inherent in the static default routed environment. VRRP specifies an election protocol that
+dynamically assigns responsibility of gateway router to a VRRP instance on one of the routers on a LAN. The VRRP instance controlling the IP address(es) associated with a virtual router is called the Master, and routes the traffic. The election process provides dynamic fail-over in the forwarding responsibility should the Master become unavailable. Any of the virtual router's IP addresses on a LAN can then be used as the default first hop router by end-hosts. The advantage gained from using VRRP is a higher availability default path without requiring configuration of dynamic routing or router discovery protocols on every end-host.
+
```

Comment:
It would be good to list the out the possible use cases for the sonic user to enable to this feature? Can this work on data center MLAG kind of deployments ?

Reply:
The use cases are networks needing backup of logical gateway. The network could be like described below or for a similar use case,
- 3-tier access layer, distribution layer & core layer network, where the redundancy for gateway is needed in distribution layer.
- Layer 3 CLOS network, where the redundancy is needed for the first hop gateways.


Generally static anycast gateway is used in MLAG deployment, but yes VRRP could also be used in MLAG deployment.